Batteries/Data/Fin/Lemmas.lean

```diff
@@ -110,11 +111,10 @@ theorem foldr_loop (f : Fin (n+1) → α → α) (x) (h : m+1 ≤ n+1) :
 theorem foldr_zero (f : Fin 0 → α → α) (x) :
     foldr 0 f x = x := foldr_loop_zero ..
 
-@[simp] theorem foldr_zero (f : Fin 0 → α → α) (x) : foldr 0 f x = x := rfl
-
 theorem foldr_succ (f : Fin (n+1) → α → α) (x) :
     foldr (n+1) f x = f 0 (foldr n (fun i => f i.succ) x) := foldr_loop ..
 
+unseal Fin.foldr.loop in
```

Comment:
No need for unsealing `Fin.foldr.loop`:
```
theorem foldr_succ_last (f : Fin (n+1) → α → α) (x) :
    foldr (n+1) f x = foldr n (f ·.castSucc) (f (last n) x) := by
  induction n generalizing x with
  | zero => simp [foldr_succ, Fin.last]
  | succ n ih => rw [foldr_succ, ih (f ·.succ), foldr_succ]; simp [succ_castSucc]
```

Reply:
I'm not sure why the `zero` case simp needs `Fin.last`. I thought `Fin.val_last` would trigger and do the right thing.